code/modules/mining/lavaland/necropolis_chests.dm

```diff
@@ -5,13 +5,52 @@
 
 /obj/structure/closet/crate/necropolis
 	name = "necropolis chest"
-	desc = "It's watching you closely."
+	desc = "It's watching you suspiciously."
 	icon_state = "necrocrate"
 	resistance_flags = LAVA_PROOF | FIRE_PROOF | ACID_PROOF
 	door_anim_time = 0
+	var/need_key = FALSE
+
+/obj/structure/closet/crate/necropolis/update_icon()
+	. = ..()
+	if(need_key == TRUE)
+		add_overlay("necrocrate_lock")
+
+/obj/structure/closet/crate/necropolis/proc/update_desc()
+	desc = "It's watching you suspiciously."
+	if(need_key == TRUE)
+		desc += " It seems like it needs a key to unlock..."
+
+/obj/structure/closet/crate/necropolis/Initialize()
+	. = ..()
+	update_desc()
+
+/obj/structure/closet/crate/necropolis/attack_hand(mob/user)
+	if(need_key == TRUE)
+		to_chat(user, "<span class='warning'>It seems that this chest is locked!</span>")
+		return
+	. = ..()
```

Comment:
A better way to implement custom-locked closets and crates would be custom togglelock proc, e.g. `/obj/structure/closet/crate/necropolis/togglelock` but I wouldn't mind if you choose not to.

Reply:
^ 
parent call should also be before everything else in attack_hand

Reply:
reworked